alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/healthprocessor/plugins/solr/SolrRequestExecutor.java

```diff
@@ -128,8 +128,27 @@ public boolean executeAsyncNodeCommand(SearchEndpoint endpoint, Status nodeStatu
         log.trace("Executing HTTP request {}", indexRequest);
         JsonNode response = httpClient.execute(indexRequest, new JSONResponseHandler());
         log.trace("Response: {}", response.asText());
+        return new SolrActionResponse(response, coreName);
+    }
+    @AllArgsConstructor
+    public static class SolrActionResponse {
+        @Getter
+        private final boolean successFull;
+        @Getter
+        private final String message;
+
+        public SolrActionResponse(JsonNode response, String coreName) {
```

Comment:
Parsing the response should happen in `executeAsyncNodeCommand`, not in a constructor.